OpenRA.Mods.Common/Orders/PlaceBuildingOrderGenerator.cs

```diff
@@ -117,6 +120,28 @@ IEnumerable<Order> InnerOrder(World world, CPos cell, MouseInput mi)
 						yield break;
 					}
 				}
+				else if (replacmentInfo != null)
+				{
+					var res = world.WorldActor.TraitOrDefault<ResourceLayer>();
+					var occupied = false;
+					foreach (var t in buildingInfo.Tiles(topLeft))
+						if (!occupied) occupied = !(world.IsCellBuildable(t, buildingInfo)
+							|| AcceptsReplacement(t, replacmentInfo))
+							|| !(res == null || res.GetResource(t) == null);
+
+					if (!buildingInfo.IsCloseEnoughToBase(world, owner, building, topLeft)
+						|| occupied)
```

Comment:
It'd be better to reorder this to `if (occupied || !buildingInfo.IsCloseEnoughToBase(world, owner, building, topLeft))` so we don't have to evaluate `IsCloseEnoughToBase` anymore if the cell is already occupied.